src/api/Main/BuisnessLayer/Shrooms.Domain/Services/Picture/PictureService.cs

```diff
@@ -34,19 +34,33 @@ public async Task<string> UploadFromImage(Image image, string mimeType, string f
         public async Task<string> UploadFromStream(Stream stream, string mimeType, string fileName, int orgId)
         {
             var pictureName = GetNewPictureName(fileName);
-            var tenantPicturesContainer = _organizationsDbSet.Where(o => o.Id == orgId).Select(o => o.ShortName).First().ToLowerInvariant();
+            var tenantPicturesContainer = await GetPictureContainer(orgId);
 
             await _storage.UploadPicture(stream, pictureName, mimeType, tenantPicturesContainer);
 
             return pictureName;
         }
 
+        public async Task RemoveImage(string blobKey, int orgId)
+        {
+            var tenantPicturesContainer = await GetPictureContainer(orgId);
+
+            await _storage.RemovePicture(blobKey, tenantPicturesContainer);
+        }
+
         private static string GetNewPictureName(string fileName)
         {
             var id = Guid.NewGuid().ToString();
             var extension = Path.GetExtension(fileName).ToLowerInvariant();
 
             return $"{id}{extension}";
         }
+
+        private async Task<string> GetPictureContainer(int id)
+        {
+            var organization = await _organizationsDbSet.FirstAsync(x => x.Id == id);
```

Comment:
Before refactoring that DB call wasn't async.
Making it async might have additional side effects, since all other DB calls here are not async.
Usually it doesn't work well with EntityFramework, when you mix sync and async DB calls..
Please test it carefully - with some different DB operations done concurrently.

Reply:
I tested it a bit and seems it works as it should, but I'm not sure if I've used a correct approach to do it.
I could make it synchronous if you wish.

Reply:
I would say - let's leave DB call synchronous.
It won't have huge improvement if we have one async call and all other calls still synchronous, but there could be issues.